packages/server/src/login/http-signature.service.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@nestjs/common';
+import * as fs from "fs";
+import * as httpSignature from 'http-signature';
+import { Request } from 'express';
+
+
+@Injectable()
+export class HttpSignatureService {
+    PUBLIC_KEY: string;
+    constructor() {
+        this.PUBLIC_KEY = fs.readFileSync(__dirname + '/khoury-public-key.pem', 'ascii');
+    }
+    public verifyRequest(req: Request): boolean {
```

Comment:
I think we should just put this in the controller. Services should deal with the business logic, but I think this is more of an HTTP-layer concern.

Reply:
Actually, leaving it in the service might make testing more feasible. I think using the `https` or `http` lib is fine - they're actually part of the node standard library so we don't need to add any packages.